plugin_export.go

```diff
@@ -13,7 +13,7 @@ import (
 func newGoPlugin(cp *C.CPlugin, c C.HostCallback) {
 	loadHook()
 	p, d := PluginAllocator(callbackHandler{c}.host(cp))
-	p.dispatchFunc = d.dispatchFunc(p.Parameters)
+	p.dispatchFunc = d.dispatchFunc(p)
```

Comment:
Could you also move it to be the last statement before mutex lock? We should add clean up to dispatcher and for that, it should have access to all things initialised/allocated in this function.